Sources/NIO/ByteBuffer-views.swift

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+public struct ByteBufferView: ContiguousCollection {
```

Comment:
This should probably implement `RandomAccessCollection` as well, to give it a bunch more utility functionality. I suspect, however, that implementing `MutableCollection` is unlikely to be a good idea: I suspect it will trigger CoWs if you actually try to mutate it.

Reply:
(and would be implied by `ContiguousCollection` I guess because I mean if you're in memory contiguously then obvs random access is fast